content/automate/ManageIQ/Transformation/Infrastructure/VM/Common.class/__methods__/poweron.rb

```diff
@@ -10,11 +10,14 @@ def initialize(handle = $evm)
               end
 
               def main
-                task = @handle.root['service_template_transformation_plan_task']
-                if task.get_option(:source_vm_power_state) == 'on'
-                  destination_vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
-                  destination_vm.start
+                if @handle.root['service_template_transformation_plan_task'].blank?
+                  task = @handle.vmdb(:service_template_transformation_plan_task).find_by(:id => @handle.root['service_template_transformation_plan_task_id'])
```

Comment:
@fdupont-redhat Check for task being nil

Reply:
I check if task is nil, as well as VM.